docs/src/design.md

```diff
@@ -30,18 +30,25 @@ One does not have to carry both a [`Model`](@ref) type, and a varying collection
 
 ```julia
 model = StatsBase.fit(
-    template,
+    template::Template,
     outputs::AbstractMatrix,  # always Features x Observations
     inputs::AbstractMatrix,   # always Variates x Observations
```

Comment:
We didn't define `input traits`. However, in BaselineModels.jl, `inputs` is an AbstractMatrix only if the model is a concrete model. If the model is a wrapper model, inputs will be a `Tuple (of Tuple(s)) AbstractMatrix`

(A good thing about this design is that, inside `fit`, we don't need to sort out inputs further for each child model. The disadvantage is memory efficiency. There are duplicated features for multiple children concrete models)

Reply:
Hmm okay. I'm not quite sure what to do about this then.

Reply:
Decision taken to leave as is, and discuss the `Tuple` interface elsewhere.